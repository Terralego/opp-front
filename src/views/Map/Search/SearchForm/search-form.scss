@import '../../../../variables.scss';

.filters {
  padding-top: 10px;

  > div {
    height: calc(100vh - 103px);
    overflow-y: auto;
    overflow-x: hidden;
  }

  .control-container {
    margin: 10px;
    padding: 5px 0 0 0;

    .tf-select {
      .bp3-popover-target {
        width: 100%;

        .bp3-button {
          width: 100%;
          text-align: left;
          justify-content: space-between;
          background-color: $darkgray;
          background-image: none;
          color: #fff;
          text-transform: lowercase;

          > span::first-letter {
            text-transform: uppercase;
          }

          &:hover {
            text-decoration: none;
            background-color: initial;
          }

          &:focus {
            box-shadow: 0 0 0 1px #137cbd, 0 0 0 3px rgba(19, 124, 189, 0.3),
              inset 0 1px 1px rgba(16, 22, 26, 0.2);
          }
        }
      }
    }

    .tf-multiselect {
      .bp3-popover-target {
        width: 100%;

        .bp3-input {
          background-color: $darkgray;
          text-transform: capitalize;

          .bp3-input-ghost {
            color: #fff;
          }
        }
      }
    }

    // Select and Multiselect Dropdown
    .bp3-menu {
      max-width: 270px;
      max-height: 300px;
      overflow: auto;
    }

    .tf-control-date-range {
      color: #000;
      padding: 0;

      .bp3-popover-target {
        width: 100%;
        background-color: $lightestgrey;
        padding: 5px;
        box-shadow: 0 0 3px rgba(255, 255, 255, 0.2) inset;

        .bp3-control-group {
          display: flex;
          flex-direction: column;
        }

        .bp3-input-group {
          border-radius: 3px;
          width: 80%;
        }
      }

      // Fix temporary
      .bp3-datepicker-caption select {
        padding-right: 0;
        padding-left: 0;
      }

      .input-range {
        display: flex;
        color: #fff;

        .bp3-input {
          background-color: $darkgray;
          color: #fff;
        }

        &:before {
          content: '';
          padding: 5px 10px 5px 5px;
          line-height: 20px;
        }
      }

      .input-range--start {
        padding-bottom: 0.5em;

        &:before {
          content: 'Du';
        }
      }

      .input-range--end {
        &:before {
          content: 'Au';
        }
      }
    }

    .bp3-input {
      background-color: $darkgray;
      color: #fff;
    }
  }

  .action-search {
    display: flex;
    width: 100%;
    justify-content: flex-end;
    padding: 10px;

    > button {
      background-color: #425360;
      background-image: none;
      margin-left: 5px;
      color: #fff;

      &:hover {
        text-decoration: none;
        background-color: #35729f;
      }

      &.bp3-disabled {
        background-color: $activetabgray;
      }
    }
  }
}
