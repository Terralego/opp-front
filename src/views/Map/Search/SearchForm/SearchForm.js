import React from 'react';
import PropTypes from 'prop-types';

import { Button } from '@blueprintjs/core';

import Filters from '@terralego/core/modules/Forms/Filters';
import { toast } from '../../../../utils/toast';
import { isDate, parsePropertiesToData } from '../../../../utils/helper/validateFormSearch';
import translateMock from '../../../../utils/translate';

import './search-form.scss';

export class SearchForm extends React.Component {
  static propTypes = {
    filters: PropTypes.arrayOf(
      PropTypes.object,
    ),
    properties: PropTypes.objectOf(
      PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
        PropTypes.array,
      ]),
    ),
    t: PropTypes.func,
  };

  static defaultProps = {
    filters: [],
    properties: {},
    t: translateMock({
      'form.reset': 'Reset',
      'form.search': 'Search',
    }),
  };

  state = {
    // Inform user of state of form : 0 = clickable, 1 = disabled
    isFormDisabled: false,
    isResetFormDisabled: false,
  };


  onChange = properties => {
    const { setProperties } = this.props;
    setProperties(properties);
  };

  onReset = async () => {
    const {
      resetMapInitialState,
      resetSearchForm,
    } = this.props;
    this.setState({ isResetFormDisabled: true });
    await resetSearchForm();
    resetMapInitialState();
    this.setState({ isResetFormDisabled: false });
  };

  onSubmit = async event => {
    event.preventDefault();
    const {
      getFirstPageFilteredViewpoints,
      forceResultUnfolding,
      itemsPerPage,
      rawData,
      properties,
    } = this.props;
    this.setState({ isFormDisabled: true });
    const data = properties ? parsePropertiesToData(properties, rawData) : {};
    const res = await getFirstPageFilteredViewpoints(data, itemsPerPage, 1);
    res ? forceResultUnfolding() : toast.displayError('Le serveur est indisponible.');
    this.setState({ isFormDisabled: false });
  };

  render () {
    const { isFormDisabled, isResetFormDisabled } = this.state;
    const { filters, properties, t } = this.props;

    const isDateInvalid = properties.viewpointDate
      && !properties.viewpointDate.every(date => isDate(date));

    return (
      <form
        className="filters"
        onSubmit={this.onSubmit}
      >
        <div>
          <Filters
            onChange={this.onChange}
            properties={properties}
            filters={filters}
            translate={t}
          />
          <div className="action-search">
            <Button
              text={t('form.reset')}
              loading={isResetFormDisabled}
              onClick={this.onReset}
            />
            <Button
              text={t('form.search')}
              type="submit"
              loading={isFormDisabled}
              disabled={isDateInvalid || isFormDisabled}
            />
          </div>
        </div>
      </form>
    );
  }
}

export default SearchForm;
