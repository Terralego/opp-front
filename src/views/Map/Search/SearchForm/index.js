import { withTranslation } from 'react-i18next';

import { connectAppProvider } from '../../../../AppLayout/context';
import { SearchForm } from './SearchForm';

export default connectAppProvider({
  properties: 'searchFormProperties',
  setProperties: 'actions.setSearchFormProperties',
  getFirstPageFilteredViewpoints: 'actions.getFirstPageFilteredViewpoints',
  forceResultUnfolding: 'actions.forceResultUnfolding',
  resetSearchForm: 'actions.resetSearchForm',
  resetMapInitialState: 'actions.resetMapInitialState',
})(withTranslation()(SearchForm));
