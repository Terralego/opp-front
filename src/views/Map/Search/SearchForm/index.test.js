import React from 'react';
import renderer from 'react-test-renderer';

import SearchForm from './SearchForm';

it('should render', () => {
  const tree = renderer.create(<SearchForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
