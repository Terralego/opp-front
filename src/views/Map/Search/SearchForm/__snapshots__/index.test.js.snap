// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<form
  className="filters"
  onSubmit={[Function]}
>
  <div>
    <div />
    <div
      className="action-search"
    >
      <button
        className="bp3-button"
        disabled={false}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="bp3-button-text"
        >
          Reset
        </span>
      </button>
      <button
        className="bp3-button"
        disabled={false}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="submit"
      >
        <span
          className="bp3-button-text"
        >
          Search
        </span>
      </button>
    </div>
  </div>
</form>
`;
